ls1intum/Artemis
# `Development`: Extract build logs from build jobs and save them in the LocalCIBuildResult in local continuous integration

### Checklist
#### General



### Motivation and Context


Currently, we aren't getting any build logs from the build jobs we run in LocalCI. This is especially bad when the build fails as we have no feedback for why this happened.

### Description

While running the script in the container, we now also save the resulting logs in a List of LogEntries. This we then pass to the `LocalCIBuildResult` class that saves it for each build.


### Steps for Testing

Prerequisites:
- 1 Students
- 1 Programming Exercise

Run a programming exercise and make it fail on purpose e.g. by removing a `}` bracket somewhere. You should receive feedback at the end why the build failed.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots


![image](https://github.com/ls1intum/Artemis/assets/120647937/7d6243f6-f936-46dd-bd28-39614b9f5aa7)
